vault/audited_headers.go

```diff
@@ -120,8 +131,15 @@ func (c *Core) setupAuditedHeadersConfig() error {
 		}
 	}
 
+	// Ensure that we are able to case-sensitively access the headers;
```

Comment:
This should not be an expensive operation given its use case. But still it seems like this `for` loop will be run again and again every time, even after the upgrade. I am not sure if having a dedicated boolean to avoid this is worth the effort. In any case, I'm inclined towards removing this code post 0.7.